safeweb/http.go

```diff
@@ -0,0 +1,89 @@
+package safeweb
+
+import (
+	crand "crypto/rand"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/csrf"
+)
+
+type Options struct {
+	Hostname                    string
+	ApplicationHandler          http.Handler
+	PermittedCrossOriginHosts   []string
+	PermittedCrossOriginMethods []string
+
+	ServeHTTPS bool
+	// ServeHTTP                   bool
+	// RedirectHTTPS               bool
```

Comment:
IMO, this should be implicitly true when `ServeHTTPS` is set or when `Server.ServeTLS(ln)` is called.
We can use the same top-level handler and check `http.Request.TLS` field to see if it's an HTTP request that needs to be redirected.